cirq-core/cirq/devices/line_qubit_test.py

```diff
@@ -287,18 +287,27 @@ def test_numeric():
     assert isinstance(complex(cirq.LineQubit(x=5)), complex)
 
 
-def test_numpy_index():
-    np5 = np.int64(5)
+@pytest.mark.parametrize('dtype', (np.int8, np.int64, float, np.float64))
+def test_numpy_index(dtype):
+    np5 = dtype(5)
     q = cirq.LineQubit(np5)
     assert hash(q) == 5
     assert q.x == 5
+    assert q.x == dtype(5)
     assert q.dimension == 2
-    assert isinstance(q.x, int)
     assert isinstance(q.dimension, int)
 
-    q = cirq.LineQid(np5, np.int64(3))
+    q = cirq.LineQid(np5, dtype(3))
     hash(q)  # doesn't throw
     assert q.x == 5
+    assert q.x == dtype(5)
     assert q.dimension == 3
-    assert isinstance(q.x, int)
     assert isinstance(q.dimension, int)
+
+
+@pytest.mark.parametrize('dtype', (float, np.float64))
+def test_non_integer_index(dtype):
```

Comment:
Maybe let's leave a comment that this is unsupported behavior that is being used anyway, just in case future changes break this again.

Reply:
Good idea. Added.